pkg/ssh/connector.go

```diff
@@ -60,7 +60,7 @@ func (c *Connector) Connect(host kubeoneapi.HostConfig) (Connection, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	conn, found := c.connections[host.ID]
+	conn, found := c.connections[host.PublicAddress]
```

Comment:
unfortunately PublicAddress is unreliable (because it's optional)

Reply:
It's the only _unique_ property of the host, and it's, as you said, always available after defaulting. So, at least for now, it works.